buyer.go

```diff
@@ -432,13 +432,22 @@ func (t *ticketPurchaser) purchase(height int64) error {
 		}
 	}
 
+	// Check for new balance credits and update ticket queue if necessary
+	if int64(balSpendable)-atomic.LoadInt64(&glBalance) > atomic.LoadInt64(&glTicketPrice) {
+		log.Tracef("Current balance %v is greater than estimated "+
+			"balance: %v", int64(balSpendable), atomic.LoadInt64(&glBalance))
```

Comment:
This isn't using atomics correctly, since it is reading the glBalance twice while expecting to see the same value.

I'm not sure without looking at other program context, but if glBalance and glTicketPrice must be updated together, then using atomics operations for each is also wrong.  You would either need to protect both with a mutex, or atomically load all of them at once (e.g. load a pointer to the most recent values saved in a heap allocation).